### PR TITLE
UI: Replace getNewModel with hydrateModel when model exists

### PR DESCRIPTION
**Pull request by hashishaw**
### Description
This PR separates logic within `getNewModel`, which previously handled both new generated models and hydrating existing models, into 2 distinct methods for each case. This is prepwork for the upcoming Ember Data upgrade. 

- [x] enterprise tests passed

### CODE REVIEW

File: ui/app/routes/vault/cluster/settings/auth/configure/section.js [hunk 1]
```diff
@@ -41,6 +42,10 @@ export default Route.extend(UnloadModelRoute, {
     const { method } = this.paramsFor('vault.cluster.settings.auth.configure');
     const backend = this.modelFor('vault.cluster.settings.auth.configure');
     const modelType = this.modelType(backend.type, section_name);
+    if (getHelpUrlForModel(modelType, 'test')) {
```
Comment (Monkeychip) [inline ui/app/routes/vault/cluster/settings/auth/configure/section.js:45]: is "test" supposed to be hardcoded arg?

File: ui/app/services/path-help.js
Comment (Monkeychip) [inline ui/app/services/path-help.js:1]: great comments 👏

File: ui/app/routes/vault/cluster/settings/auth/configure/section.js (hunk 1 above)
Comment (hashishaw) [reply ui/app/routes/vault/cluster/settings/auth/configure/section.js:45]: It actually doesn't have to be anything, I'm using this function as a test to see if the given model has a helpUrl defined (basically the same as previously checking model.useOpenApi). I'll remove it and add a comment to what this is doing.

**Review by Monkeychip — APPROVED** [review]
🚀

**Pull request merged**